code/modules/client/preference_setup/general/01_basic.dm

```diff
@@ -51,6 +51,25 @@
 				":id" = pref.current_character,
 				":ckey" = pref.client.ckey)
 
+/datum/category_item/player_setup_item/general/basic/load_special()
+	pref.can_edit_name = 1
+
+	if (config.sql_saves && pref.current_character)
+		if (!establish_db_connection(dbcon))
+			return
+
+		// Called /after/ loading and /before/ sanitization.
+		// So we have pref.current_character. It's just in text format.
+		var/DBQuery/query = dbcon.NewQuery("SELECT DATEDIFF(NOW(), created_at) AS DiffDate FROM ss13_characters WHERE id = :id")
+		query.Execute(list(":id" = text2num(pref.current_character)))
+
+		if (query.NextRow())
+			if (text2num(query.item[1]) > 5)
```

Comment:
I would add two config items:
* To enable/disable this check
* To set the time of days a user can edit his char